lib/fizzy/execute.cpp

```diff
@@ -140,14 +140,15 @@ inline uint32_t grow_memory(
 {
     const auto cur_pages = memory.size() / PageSize;
```

Comment:
This function looks like a mess now with assertions every second line. Not sure yet, but can we somehow clean this mess up a bit?

Reply:
Also while we're speaking assertions, there could be one for `memory.size() % PageSize == 0` :wink:

Reply:
I tried to clean up by making assertions simpler and adding some comments.